etudes-francaises.csl

```diff
@@ -0,0 +1,414 @@
+            <text variable="publisher-place"/>
+            <text variable="publisher"/>
+            <text macro="year-date"/>
+            <text macro="number-of-volumes"/>
+            <text macro="number-of-pages"/>
+          </group>
+        </if>
+        <else-if type="entry-dictionary entry-encyclopedia" match="any">
+          <group delimiter=", ">
+            <group delimiter=" ">
+              <text variable="container-title" font-style="italic" text-case="title" form="short"/>
+              <number variable="volume" form="roman"/>
+            </group>
+            <text macro="year-date"/>
+            <group delimiter=" ">
+              <text term="column" form="short"/>
+              <text variable="page"/>
+            </group>
+            <group delimiter=" ">
+              <text macro="title" quotes="true"/>
+            </group>
+          </group>
+        </else-if>
+        <else-if type="webpage article-journal article-magazine article-newspaper broadcast personal_communication article" match="any">
+          <group delimiter=", ">
+            <text macro="title" quotes="true"/>
+            <text variable="container-title" form="short" font-style="italic" text-case="title"/>
+            <text macro="volume"/>
+            <text macro="issue"/>
+            <group delimiter=" ">
+              <text macro="year-date"/>
+              <text macro="edition"/>
+            </group>
+            <text macro="pages"/>
+          </group>
+        </else-if>
+        <else-if type="book">
+          <choose>
+            <if variable="author">
+              <choose>
+                <if variable="editor translator" match="none">
+                  <text macro="book"/>
+                </if>
+                <else>
+                  <group delimiter=", ">
+                    <text macro="title" font-style="italic"/>
+                    <text macro="editor-translator"/>
+                    <text macro="publisher"/>
+                    <text macro="collection"/>
+                    <group delimiter=" ">
+                      <text macro="year-date"/>
+                      <text macro="edition"/>
+                    </group>
+                  </group>
+                </else>
+              </choose>
+            </if>
+            <else>
+              <text macro="book"/>
+            </else>
+          </choose>
+        </else-if>
+        <else-if type="chapter paper-conference" match="any">
+          <group delimiter=", ">
+            <text macro="title" quotes="true"/>
+            <choose>
+              <if variable="editor container-author" match="any">
+                <group>
+                  <text term="in" suffix=" "/>
+                  <text macro="editor" suffix=", "/>
+                  <text variable="container-title" font-style="italic" text-case="title"/>
+                </group>
+              </if>
+              <else>
+                <group>
```

Comment:
We always recommend using delimiters instead of pre/suffix as it can prevent formatting errors.
E.g. this section would be better like this:
`               <group delimiter=" ">
                  <text term="in"/>
                  <text variable="container-title" font-style="italic" text-case="title"/>
                </group>`

Reply:
Thank you. However, the example you are providing is missing the `editor` part, which requires a comma separator. In addition to the group delimiter, should I use a suffix for this particular comma, e.g. lines 320-324:

```csl
h="any">
<group delimiter=" ">
  <text term="in"/>
  <text macro="editor" suffix=","/>
  <text variable="container-title" font-style="italic" text-case="title"/>
</group>
```

Reply:
No, see which lines this refers to. I copied the code from lines 327-330, not the section above.

Reply:
Should be resolved now.